### DISCUSSION TITLE
GX ESS Scheduled Chare Levels

### DISCUSSION THREAD

**Discussion by DuncanGardiner** [sfstar/hass-victron#215] [Ideas]
Hi I am based in Portugal and have a Multiplus ii 48/8000 setup with a ESS assistant. The Ess assistant allows my system to be connected to the grid to push or pull energy from my supplier.  
I normally only push excess energy to the supplier and run my loads in the house for the most part in an off grid state (via a battery bank). 
On occasion when the batteries get very low due to rain days or heavy cloud. I use the scheduled charged option within ESS section of the GX to charge my batteries from the grid at night (thus buying power when its cheaper here). 

My question is:

Is the toggle for the schedules (the GX has 5 schedules that you can set) within the ESS section exposed via Mod-Bus-TCP, and can I interact withg them from HA. 
I'd like to build an autmation that activates a scheduled charge or deactivates a scheduled charge when conditions within HA are met?   

Kind regards Duncan

**Comment by pos-ei-don** [comment]
I use the "min soc" setting for this. Incressing this to 60 or 100% starts charging the battery. Works works great, ana can also be variable when using hourly based grid Tarifs.

**Reply by DuncanGardiner** [reply]
Hi, yes I have seen a few people use that setting, in combination with the aux ports. 

I was hoping that I could use the 'Schedule Charge Level' toggles within the GX as that would give me a wider range of preset charging times within the Cebro GX it self. I am trying to aviod having schedules for charging within HA, as I can switch the GX schedules on and off in software remotely when I am not home, where as my HA echo system is on a local LAN only.

Kind regards

Duncan
